python_ta/reporters/node_printers.py

```diff
@@ -153,6 +205,328 @@ def render_blank_line(line):
+        slice(0, msg_line_start_index),
+        LineType.ERROR,
+        source_lines[msg.line - 1],
+    )
+    yield from render_context(msg.line + 1, msg.line + 3, source_lines)
+
+
+def render_pep8_errors_e122_and_e127_and_e131(msg, _node, source_lines=None):
+    """
+    Render a PEP8 continuation line missing indentation or outdented message, a line over-indented for visual indent
+    message, and a continuation line unaligned for hanging indent message.
+    """
+    line = msg.line - 1
+    curr_line_start_index = 0
+    while source_lines[line][curr_line_start_index] == " ":
+        curr_line_start_index += 1
+    end_index = curr_line_start_index if curr_line_start_index > 0 else len(source_lines[line])
+    yield from render_context(line - 1, line + 1, source_lines)
+    yield (
+        msg.line,
+        slice(0, end_index),
+        LineType.ERROR,
+        source_lines[line],
+    )
+    yield from render_context(msg.line + 1, msg.line + 3, source_lines)
+
+
+def render_pep8_errors_e124(msg, _node, source_lines=None):
+    """Render a PEP8 closing bracket does not match visual indentation message."""
+    line = msg.line - 1
+    res = re.search(r"column (\d+)", msg.msg)
+    col = int(res.group().split()[-1])
+    yield from render_context(line - 1, line + 1, source_lines)
+    yield (msg.line, slice(col, col + 1), LineType.ERROR, source_lines[line])
+    yield from render_context(msg.line + 1, msg.line + 3, source_lines)
+
+
+def render_pep8_errors_e125_and_e129(msg, _node, source_lines=None):
+    """Render a PEP8 continuation line with same indent as next logical line message
+    AND a PEP8 visually indented line with same indent as next logical line messsage"""
+    msg_line_start_index = 0
+
+    while source_lines[msg.line - 1][msg_line_start_index] == " ":
+        msg_line_start_index += 1
+
+    yield from render_context(msg.line - 2, msg.line, source_lines)
+    yield (
+        msg.line,
+        slice(msg_line_start_index, len(source_lines[msg.line - 1])),
+        LineType.ERROR,
+        source_lines[msg.line - 1] + " " * 2 + "# INDENT THIS LINE",
+    )
+    yield from render_context(msg.line + 1, msg.line + 3, source_lines)
+
+
+def render_pep8_errors_e128(msg, _node, source_lines):
+    """Render a PEP8 continuation line under-indented for visual indent message."""
+    line = msg.line - 1
+    res = re.search(r"column (\d+)", msg.msg)
+    col = int(res.group().split()[-1])
+
+    yield from render_context(line - 1, line + 1, source_lines)
+    yield (msg.line, slice(0, col if col != 0 else None), LineType.ERROR, source_lines[line])
+    yield from render_context(msg.line + 1, msg.line + 3, source_lines)
+
+
+def render_pep8_errors_e201(msg, _node, source_lines=None):
+    """Render a PEP8 whitespace after '(' message."""
+    line = msg.line - 1
+    res = re.search(r"column (\d+)", msg.msg)
+    col = int(res.group().split()[-1])
+    curr_idx = col
+    while source_lines[line][curr_idx].isspace():
```

Comment:
use can use `str.lstrip` here as well, you just need to do it on `source_lines[line][curr_idx:]`